### PR TITLE
Fix for Foxx Service Installation

### PR DESCRIPTION
**Pull request by hkernbach**
### Scope & Purpose

This PR fixes the installation of Foxx Services in case a LoadBalancer is sitting in front of the coordinator.
If a coordinator is now asked to install a zip file - which has not been uploaded to itself - it sends the request to the correct coordinator using the correct coordinator id. 

- [x] :hankey: Bugfix

### Checklist

- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.10: *(Please link PR)*
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*

#### Related Information

- [x] GitHub issue / Jira ticket: https://arangodb.atlassian.net/browse/BTS-1315

### CODE REVIEW

**Comment by joerg84** [comment]
@hkernbach is this ready to be merged

**Review by markuspf — CHANGES_REQUESTED** [review]
I think the fix is OK, but we could add a deterministic test that uploads a zip to one coordinator and then tries sending an install request to a different one.

This should be always red before this fix, and always green after, giving slightly more confidence that you fixed something.

File: js/apps/system/_admin/aardvark/APP/foxxes.js [hunk 1]
```diff
@@ -90,13 +92,32 @@ installer.use(function (req, res, next) {
   options.setup = req.queryParams.setup;
   options.teardown = req.queryParams.teardown;
   let service;
+
+  let continueFoxxOperation = true;
+  const thisCoordinatorId = FoxxManager._getMyCoordinatorId();
+  if (thisCoordinatorId) {
+    if (req.body.coordinatorId) {
```
Comment (markuspf) [inline js/apps/system/_admin/aardvark/APP/foxxes.js:99]: What happens if you (maliciously) give a non-existent coordinator ID in this request?

**Comment by markuspf** [comment]
I also "verified" manually that I could not reproduce the wrong behaviour on an ArangoGraph deployment.

File: js/apps/system/_admin/aardvark/APP/foxxes.js (hunk 1 above)
Comment (hkernbach) [reply js/apps/system/_admin/aardvark/APP/foxxes.js:101]: https://github.com/arangodb/arangodb/pull/18691/files#diff-970983882d6f4ad739c362b38dac0ef2d7cd1aa12cf57f024d31c74125c3f782R239-R246

Now being handled here.

**Comment by hkernbach** [comment]
@markuspf added LoadBalancer tests and will now re-request a review :P

File: tests/js/client/load-balancing/load-balancing-foxx-installation-noauth-cluster.js [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+let coordinators = []; // will be initialized during setup
+const installMountPath = `/aRandomMountPathName`;
+const databaseName = '_system';
+
+const loadFoxxIntoZip = (path) => {
+  let zip = utils.zipDirectory(path);
+  let content = fs.readFileSync(zip);
+  fs.remove(zip);
+  //Content-Disposition: form-data; name="file";
```
Comment (hkernbach) [inline tests/js/client/load-balancing/load-balancing-foxx-installation-noauth-cluster.js:48]:
```suggestion
```

**Review by markuspf — APPROVED** [review]

**Pull request merged**